apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/TransactionAwareSubscriber.java

```diff
@@ -0,0 +1,385 @@
+
+        // In case transaction has been created by Servlet, we should not terminate it as the Servlet instrumentation
+        // will take care of this.
+        if (!WebfluxHelper.isServletTransaction(exchange)) {
+            transaction.end();
+        }
+
+    }
+
+    private static boolean ignoreTransaction(ServerWebExchange exchange, Transaction transaction) {
+        // Annotated controllers have the invoked handler method available in exchange
+        // thus we can rely on this to ignore methods that return ServerSideEvents which should not report transactions
+        Object attribute = exchange.getAttribute(HandlerMapping.BEST_MATCHING_HANDLER_ATTRIBUTE);
+        if (!(attribute instanceof HandlerMethod)) {
+            return false;
+        }
+
+        HandlerMethod handlerMethod = (HandlerMethod) attribute;
+        Boolean ignoredCache = ignoredHandlerMethods.get(handlerMethod);
+        if (ignoredCache != null) {
+            return ignoredCache;
+        }
+
+        Type returnType = handlerMethod.getMethod().getGenericReturnType();
+        if (!(returnType instanceof ParameterizedType)) {
+            ignoredHandlerMethods.put(handlerMethod, false);
+            return false;
+        }
+
+        Type[] genReturnTypes = ((ParameterizedType) returnType).getActualTypeArguments();
+        //noinspection ForLoopReplaceableByForEach
+        for (int i = 0; i < genReturnTypes.length; i++) {
+            if (genReturnTypes[i].getTypeName().startsWith(WebfluxHelper.SSE_EVENT_CLASS)) {
+                ignoredHandlerMethods.put(handlerMethod, true);
+                return true;
+            }
+        }
+
+        ignoredHandlerMethods.put(handlerMethod, false);
+        return false;
+    }
+
+    private static void fillRequest(Transaction transaction, ServerWebExchange exchange) {
+        ServerHttpRequest serverRequest = exchange.getRequest();
+        Request request = transaction.getContext().getRequest();
+
+        request.withMethod(serverRequest.getMethodValue());
+
+        InetSocketAddress remoteAddress = serverRequest.getRemoteAddress();
+        request.getSocket()
+            .withRemoteAddress(remoteAddress == null ? null : remoteAddress.getAddress().getHostAddress())
+            .withEncrypted(serverRequest.getSslInfo() != null);
+
+        URI uri = serverRequest.getURI();
+        request.getUrl()
+            .withProtocol(uri.getScheme())
+            .withHostname(uri.getHost())
+            .withPort(uri.getPort())
+            .withPathname(uri.getPath())
+            .withSearch(uri.getQuery())
+            .updateFull();
+
+        copyHeaders(serverRequest.getHeaders(), request.getHeaders());
+
+        for (Map.Entry<String, List<HttpCookie>> cookie : serverRequest.getCookies().entrySet()) {
+            for (HttpCookie value : cookie.getValue()) {
+                request.getCookies().add(cookie.getKey(), value.getValue());
+            }
+        }
+
```

Comment:
How complicated would it be to use `serverRequest.getBody()` to capture the request body?

Reply:
I don't think it would be easy to capture body using this, as we get a `Flux` and not a direct access to the request body byte buffer. Would better off keeping that for a follow-up PR.